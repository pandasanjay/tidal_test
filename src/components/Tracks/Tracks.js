import React from "react";
import { fmtMSS } from "../../utils/functionUtils";
const Tracks = ({ lists = [], releaseyear = "" }) => {
  return (
    <>
      <table>
        <thead>
          <tr>
            <th>#</th>
            <th>Title</th>
            <th>Artist</th>
            <th>Time</th>
            <th>Released</th>
          </tr>
        </thead>
        <tbody>
          {lists.map((data, i) => (
            <tr key={i}>
              <td>{i + 1}</td>
              <td>{data.title}</td>
              <td>{data.artist.name}</td>
              <td>{fmtMSS(data.duration)}</td>
              <td>{releaseyear}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </>
  );
};

export default Tracks;
