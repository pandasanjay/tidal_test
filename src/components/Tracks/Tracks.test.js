import React from "react";
import { render } from "@testing-library/react";
import Tracks from "./Tracks";
describe("AlbumDetails connected component ", () => {
  it("should render and check the snapshot", () => {
    const stubLists = [
      {
        title: "foo",
        artist: {
          name: "bar"
        },
        duration: 422
      }
    ];
    const { asFragment } = render(
      <Tracks lists={stubLists} releaseyear="2019" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
