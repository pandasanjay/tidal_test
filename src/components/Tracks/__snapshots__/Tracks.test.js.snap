// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumDetails connected component  should render and check the snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Title
        </th>
        <th>
          Artist
        </th>
        <th>
          Time
        </th>
        <th>
          Released
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          foo
        </td>
        <td>
          bar
        </td>
        <td>
          7:02
        </td>
        <td>
          2019
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
